selfdrive/debug/toyota_eps_factor.py

```diff
@@ -2,26 +2,32 @@
 import sys
 import numpy as np
 import matplotlib.pyplot as plt
-from sklearn import linear_model # pylint: disable=import-error
+from sklearn import linear_model  # pylint: disable=import-error
 
 from tools.lib.route import Route
 from tools.lib.logreader import MultiLogIterator
 
+MIN_SAMPLES = 30 * 100
 
-MIN_SAMPLES = 30*100
 
 def to_signed(n, bits):
   if n >= (1 << max((bits - 1), 0)):
     n = n - (1 << max(bits, 0))
   return n
 
+
 def get_eps_factor(lr, plot=False):
+  all_msgs = sorted(lr, key=lambda msg: msg.logMonoTime)
 
   engaged = False
+  steering_pressed = False
   torque_cmd, eps_torque = None, None
   cmds, eps = [], []
 
-  for msg in lr:
+  for msg in all_msgs:
+    if msg.which() == 'carState':
+      steering_pressed = msg.carState.steeringpressed
+
```

Comment:
Is this the correct way to get the `steeringPressed` message when we're at a CAN message? Basically cache them into a variable and just reference it every time we're in a CAN message?

Reply:
ideally, all of this should just use the can parser, but then you have to account for the factor of the DBC you use since you want the raw signal. i'd just parse out steeringPressed from the can msg instead of using carState.

Reply:
How would I get that info from the can signal? if `STEER_TORQUE_DRIVER` is > 100?

Reply:
yeah, that's how we do it in carstate

Reply:
Okay, I was just concerned that the way it's looping through all the messages one by one and not as frames that the last carState message might be out of sync when we're at the CAN data. I'm not really sure how to parse the binary CAN data but I'll try to at least start on getting it from the CAN message